skbio/sequence/_substitution.py

```diff
@@ -0,0 +1,518 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import numpy as np
+
+from skbio.util._decorator import experimental, classonlymethod
+from skbio.stats.distance import DissimilarityMatrix
+
+
+@experimental(as_of='0.5.10')
+class SubstitutionMatrix(DissimilarityMatrix):
+    """Matrix for scoring substitutions between characters in biological
+    sequences.
```

Comment:
I'm surprised this made it through lint. PEP8 extends PEP257, which implies use of a single line summary, see [here](https://peps.python.org/pep-0257/#multi-line-docstrings). That said, I think we should definitely switch to `black` or `ruff` and just never talk about style again :)

```suggestion
    """Scoring matrix between characters in biological sequences
```

Reply:
Great to know! I wasn't aware of it. I will try to make single-line summaries in the future.